Detectors/TOF/simulation/include/TOFSimulation/Digitizer.h

```diff
@@ -77,18 +85,41 @@ class Digitizer
   Float_t mEffBoundary3;
 
   // info TOF timewindow
-  Int_t mTimeFrameCurrent;
+  Int_t mReadoutWindowCurrent;
   Double_t mEventTime;
   Int_t mEventID = 0;
   Int_t mSrcID = 0;
 
+  bool mContinuous = false;
+
   // digit info
-  std::vector<Digit>* mDigits;
-  o2::dataformats::MCTruthContainer<o2::tof::MCLabel>* mMCTruthContainer =
-    nullptr; ///< Array for MCTruth information associated to digits in mDigitsArrray. Passed from the digitization
+  //std::vector<Digit>* mDigits;
+
+  static const int MAXWINDOWS = 10; // how many readout windows we can buffer
+
+  int mIcurrentReadoutWindow = 0;
+  o2::dataformats::MCTruthContainer<o2::tof::MCLabel> mMCTruthContainer[MAXWINDOWS];
+  o2::dataformats::MCTruthContainer<o2::tof::MCLabel>* mMCTruthContainerCurrent = &mMCTruthContainer[0]; ///< Array for MCTruth information associated to digits in mDigitsArrray.
+  o2::dataformats::MCTruthContainer<o2::tof::MCLabel>* mMCTruthContainerNext[MAXWINDOWS - 1];            ///< Array for MCTruth information associated to digits in mDigitsArrray.
+  o2::dataformats::MCTruthContainer<o2::MCCompLabel>* mMCTrueContainer;
```

Comment:
is it intentional to use `o2::MCCompLabel` as well as `o2::tof::MCLabel` at the same time?
What is the difference? Maybe one could also choose slightly more distinguishable names rather than `mMCTrueContainer` vs `mMCTruthContainer`

Reply:
mMCTruthContainer is the array of TOF labels, where in addition to the MCCompLabel we have the tdc info. 
mMCTrueContainer is the array of labels that we write in the output. 
The names are indeed an unfortunate choice, we changed mMCTrueContainer to mMCTruthOutputContainer.